patches/@nx+esbuild+17.0.3.patch

```diff
@@ -0,0 +1,41 @@
+diff --git a/node_modules/@nx/esbuild/src/executors/esbuild/lib/build-esbuild-options.js b/node_modules/@nx/esbuild/src/executors/esbuild/lib/build-esbuild-options.js
+index 7f016a3..bf46353 100644
+--- a/node_modules/@nx/esbuild/src/executors/esbuild/lib/build-esbuild-options.js
++++ b/node_modules/@nx/esbuild/src/executors/esbuild/lib/build-esbuild-options.js
+@@ -98,23 +98,26 @@ function buildEsbuildOptions(format, options, context) {
+ }
+ exports.buildEsbuildOptions = buildEsbuildOptions;
+ function getOutExtension(format, options) {
+-    const userDefinedExt = options.userDefinedBuildOptions?.outExtension?.['.js'];
++    const userDefExts = options.userDefinedBuildOptions?.outExtension ?? {};
++
+     // Allow users to change the output extensions from default CJS and ESM extensions.
+     // CJS -> .js
+     // ESM -> .mjs
+-    return userDefinedExt === '.js' && format === 'cjs'
+-        ? '.js'
+-        : userDefinedExt === '.mjs' && format === 'esm'
+-            ? '.mjs'
+-            : format === 'esm'
+-                ? ESM_FILE_EXTENSION
+-                : CJS_FILE_EXTENSION;
++    const defaultExt = format === 'cjs' ? CJS_FILE_EXTENSION : format === 'esm' ? ESM_FILE_EXTENSION : '.js';
++    if (userDefExts[defaultExt]) {
++        return userDefExts[defaultExt];
++    }
++
++    return defaultExt;
```

Comment:
Is nx esbuild overwriting the path extensions?

Reply:
So it had some dubious logic (being patched here) about how to derive an extension, even if the configured `outputFileName` had a valid extension - so the patch:
1) Doesn't calculate a new extension if `outputFileName` already has an extension of `js`, `cjs` or `mjs`.
2) Uses the user-configured extension mapping in the esbuild options if they are configured.
3) Only then falls back to the defaults the Nx plugin decides.